nifi-nar-bundles/nifi-extension-utils/nifi-database-utils/src/main/java/org/apache/nifi/util/db/JdbcCommon.java

```diff
@@ -807,7 +807,7 @@ public static void setParameter(final PreparedStatement stmt, final String attrN
                             throw new ParseException("Unable to parse binary data using the formatter `" + valueFormat + "`.",0);
                     }
 
-                    stmt.setBinaryStream(parameterIndex, new ByteArrayInputStream(bValue), bValue.length);
+                    stmt.setBytes(parameterIndex, bValue);
```

Comment:
How about we try the `setBinaryStream` first, then if it throws a `SQLFeatureNotSupportedException` we try `setBytes`? This might prevent a regression if there is some other driver (not Phoenix) that supports setBinaryStream and not setBytes. This code is used by many folks with many different drivers/DBs, so I'd like to minimize the chance for regressions. Thoughts?

Reply:
I agree, keeping the old one with a failover would be a better solution. The only thing that sucks is you'll be catching the exception for every row, and then switching to the alternate. I suppose alternatively you could put a flag, and set the flag on first exception...

Reply:
@mattyb149 @patricker I'm not a fan of setting a flag because after skimming over that method I didn't see a good place to store a state to track whether the except was thrown in a previous pass. I'll add the try/catch and merge.

Reply:
Hmm.. what about adding an upstream check to see if PreparedStatement instanceof PhoenixPreparedStatement, which can be handled independently? Using try/catch as control flow doesn't seem quite right, esp. for something that supports operations that need to be very performant (e.g. data ingestion). This would also give us a pattern to handle other JDBC drivers that are not totally compliant and handle the case where developers wish to support two drivers that are incompatible (for example, one that only has .setBytes(...) _and_ one that only has .setBinaryStream(...)

Reply:
That would require us to have a compile time dependency on the phoenix driver to be able to reference `PhoenixPreparedStatement` with `instanceof`.